subsys/net/ip/Kconfig.tcp

```diff
@@ -66,14 +66,14 @@ config NET_TCP_TIME_WAIT_DELAY
 	  Value of 0 disables TIME_WAIT state completely.
 
 config NET_TCP_ACK_TIMEOUT
-	int "How long to wait for ACK (in milliseconds)"
+	int "[DEPRECATED] How long to wait for ACK (in milliseconds)"
 	depends on NET_TCP
 	default 1000
 	range 1 2147483647
```

Comment:
> We could also add select DEPRECATED here.

That won't work, since the Kconfig is `int` type, it cannot select anything.

> As we basically made the variable no-op, should we remove it altogether. I mean it is not really deprecated in this case if it cannot be used at all.

I don't personally have a problem with that, but wouldn't that violate our API Lifecycle guidelines?

Reply:
I don't mind leaving the option here if needed. I just thought the deprecation means that the old behavior works but gives a warning so that user can avoid using it at some point. But using this deprecated symbol cannot be really used after this change, anyway perhaps it makes sense to inform user to use some other config symbol like done here.